core/trino-main/src/main/java/io/trino/execution/scheduler/FaultTolerantStageScheduler.java

```diff
@@ -400,14 +425,40 @@ private void startTask(int partition, NodeAllocator.NodeLease nodeLease, MemoryR
         task.start();
     }
 
+    private void createOutputSelectorSplitsIfNecessary()
+    {
+        if (outputSelectorSplits != null) {
```

Comment:
![image](https://user-images.githubusercontent.com/890111/193770713-47637c64-034b-4c9e-821a-8b00d640a505.png)

please enable intellij warning

Reply:
As a fix in this case you should just add
`@GuardedBy("this")` on `startTask`

Reply:
That's probably an enterprise version feature. Added an annotation

Reply:
Oh - interesting. This one is pretty useful actually

Reply:
You also need that on `startTask` I think